# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: accessible/mac/MOXTextMarkerDelegate.mm
# Commit: 6ff900a299dc
# Full Hash: 6ff900a299dce4f3c02fc2946d91d5514e460f49
# Author: Eitan Isaacson <eitan@monotonous.org>
# Date: 2020-07-23 03:20:25
# Description:
#   Bug 1654601 - Ensure given text marker is valid in public API. r=MarcoZ
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D84567
# ==============================================================================

diff -r f1a36fd1fac8 -r 6ff900a299dc accessible/mac/MOXTextMarkerDelegate.mm
--- a/accessible/mac/MOXTextMarkerDelegate.mm	Wed Jul 22 17:51:14 2020 +0000
+++ b/accessible/mac/MOXTextMarkerDelegate.mm	Wed Jul 22 18:08:10 2020 +0000
@@ -143,6 +143,10 @@
 
 - (id)moxLeftWordTextMarkerRangeForTextMarker:(id)textMarker {
   GeckoTextMarker geckoTextMarker(mGeckoDocAccessible, textMarker);
+  if (!geckoTextMarker.IsValid()) {
+    return nil;
+  }
+
   geckoTextMarker.NormalizePrevious();
 
   if (geckoTextMarker.mOffset == 0) {
@@ -160,6 +164,10 @@
 
 - (id)moxRightWordTextMarkerRangeForTextMarker:(id)textMarker {
   GeckoTextMarker geckoTextMarker(mGeckoDocAccessible, textMarker);
+  if (!geckoTextMarker.IsValid()) {
+    return nil;
+  }
+
   geckoTextMarker.NormalizeNext();
 
   GeckoTextMarkerRange range = geckoTextMarker.AtEnd()
@@ -171,6 +179,9 @@
 
 - (id)moxNextTextMarkerForTextMarker:(id)textMarker {
   GeckoTextMarker geckoTextMarker(mGeckoDocAccessible, textMarker);
+  if (!geckoTextMarker.IsValid()) {
+    return nil;
+  }
 
   geckoTextMarker.NormalizeNext();
   if (geckoTextMarker.AtEnd()) {
@@ -184,6 +195,9 @@
 
 - (id)moxPreviousTextMarkerForTextMarker:(id)textMarker {
   GeckoTextMarker geckoTextMarker(mGeckoDocAccessible, textMarker);
+  if (!geckoTextMarker.IsValid()) {
+    return nil;
+  }
 
   geckoTextMarker.NormalizePrevious();
   if (geckoTextMarker.mOffset == 0) {
